wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt

```diff
@@ -64,6 +64,33 @@ sealed class Target {
   interface SchemaHandler {
     fun handle(type: Type)
     fun handle(service: Service)
+    /**
+     * This will handle all [Type]s and [Service]s of the `protoFile` in respect to the emitting
+     * rules. If exclusive, the handled [Type]s and [Service]s should be added to the consumed set.
+     * Consumed types and services themselves are to be omitted by this handler.
+     */
+    fun handle(
+      protoFile: ProtoFile,
+      emittingRules: PruningRules,
+      consumedTypesAndServices: MutableSet<Any>,
```

Comment:
I’d prefer a purpose-built type for this

Reply:
will do in follow up